### PR TITLE
Softcode Fountain of Youth

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
... Kilimanjaro should now be only a matter of json. Not that we want it now ...
- I did _not_ attempt to fix the deviation that if I remember correctly still exists from our template game: Fountain of Youth should give +10 Happy when owned, but cannot be worked by a city? Something like that.

Only some very quick testing so far.

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/unique/UniqueParameterType.kt [hunk 1]
```diff
@@ -18,6 +18,10 @@ enum class UniqueParameterType(val parameterName:String) {
         private val knownValues = setOf("Wounded", "Barbarians", "City-State", "Embarked", "Non-City")
         override fun getErrorSeverity(parameterText: String, ruleset: Ruleset):
                 UniqueType.UniqueComplianceErrorSeverity? {
+            if ('{' in parameterText) // "{filter} {filter}" for and logic
+                return parameterText.removePrefix("{").removeSuffix("}").split("} {")
```
Comment (xlenstra) [inline core/src/com/unciv/models/ruleset/unique/UniqueParameterType.kt:22]: Does this still work for languages that don't use spaces?

Comment (SomeTroglodyte) [reply core/src/com/unciv/models/ruleset/unique/UniqueParameterType.kt:22]: And I thought that code I copied there was yours... But no, blame says Yair's.
This is just a counterpart to MapUnit.matchesFilter so the unique-auto-checks will accept the same input as the filter itself.
And this is pre-translation, so... Should not be a step back? But that's why I value input - sometimes I get the feeling to have missed some aspect and can't put my finger on it, this one especially so.

**Comment by SomeTroglodyte** [comment]
'nuff testing. Most relevant result from testing - Those 3-4 "Double movement" uniques need merging, which emphasizes again some thinking about and/or for terrainFilters would be nice.

File: core/src/com/unciv/models/ruleset/unique/UniqueParameterType.kt (hunk 1 above)
Comment (yairm210) [reply core/src/com/unciv/models/ruleset/unique/UniqueParameterType.kt:22]: This looks good to me!
The translation of said parameters should be dealt with, but definitely outside this PR

**Pull request merged**